src/components/schedule-page/toolbar/toolbar-viewing.component.tsx

```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import { Button } from "../../common-components";
+import { ImportButtonsComponent } from "../import-buttons/import-buttons.component";
+import RouteButtonsComponent from "../../common-components/route-buttons/route-buttons.component";
+import SchedulePage from "../schedule-page.component";
+import ManagementPage from "../../workers-page/management-page.component";
+import { Box } from "@material-ui/core";
+
+interface TabData {
+  label: string;
+  component: JSX.Element;
+  right_side_buttons: JSX.Element;
+}
+
+export function ToolbarViewingComponent(props): JSX.Element {
+  const handleEditMode = props.editModeChange;
+
+  function toggleEdit(open: boolean): void {
+    handleEditMode(open);
+  }
+
+  function Buttons(): JSX.Element {
+    return (
+      <>
+        <div className={"buttons"}>
+          <ImportButtonsComponent />
+          <Box>
+            <Button
+              size="small"
+              className="submit-button"
+              variant="primary"
+              onClick={() => toggleEdit(true)}
+            >
+              Edytuj
+            </Button>
+          </Box>
+        </div>
+      </>
+    );
+  }
+
+  const tabs: TabData[] = [
+    // eslint-disable-next-line @typescript-eslint/camelcase
+    { label: "Plan", component: <SchedulePage />, right_side_buttons: Buttons() },
```

Comment:
Szczerze mówiąc nie pamiętam dlaczego, ale lepiej nie tworzyć componentów w taki sposób `Buttons()`. Zamiast tego należy użyć składni `<Buttons/>`. To jest związane z tym, że te nawiasy `</>` - t otak naprawdę cukier składniowy, który zamienia `React.createElement` w coś bardziej czytelne dla człowieka. 
To jest w jakiś sposób związany z hookami, czyli tym jak react resjesrtruje komponenty które były utworzone, ale niestety nie udało mi się tego odtworzyć.